kahypar/datastructure/hypergraph.h

```diff
@@ -606,10 +618,12 @@ class GenericHypergraph {
                     const HypernodeID* edge_vector,
                     const PartitionID k = 2,
                     const HyperedgeWeight* hyperedge_weights = nullptr,
-                    const HypernodeWeight* hypernode_weights = nullptr) :
+                    const HypernodeWeight* hypernode_weights = nullptr,
+                    const std::vector<HyperedgeID>& ignored_hyperedges = {},
```

Comment:
I might be missing something, but wouldn't it be possible to move the handling of `ignored_hyperedges` and `ignored_pins` upstream of the hypergraph construction such that the constructor just always gets valid `index`, `edge`, and `weights` vectors?

Reply:
@SebastianSchlag any thoughts on this issue?

Reply:
>  I intentionally avoided this in order to minimize the runtime impact of the input validation. But if you think this is not an issue I can move it out of the constructor.

Thanks for the explanation.
I think the overhead for copying should be acceptable and this would make the interface much cleaner.

Reply:
Actually, the more I think about it... I tend to agree that your approach is actually fine - no need to change it.

Reply:
Great :+1: